include/bluetooth/services/gadgets.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GADGETS_H_
+#define BT_GADGETS_H_
+
+/**
+ * @file
+ * @defgroup bt_gadgets Alexa Gadgets GATT Service
+ * @{
+ * @brief Alexa Gadgets GATT Service API.
+ * @details Implements the Alexa Gadgets GATT Service,
+ *          and handles the data format (streams) that wraps the data payload
+ *          sent over this Service.
+ *          See Alexa Gadgets documentation for further information:
+ *          https://developer.amazon.com/en-US/docs/alexa/alexa-gadgets-toolkit/packet-ble.html
+ */
+
+#include <zephyr/types.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief UUID of the Gadgets Service. */
+#define GADGETS_UUID_SERVICE \
+	BT_UUID_128_ENCODE(0x0000FE03, 0x0000, 0x1000, 0x8000, 0x00805F9B34FB)
+
+/** @brief Short version of the Gadgets Service UUID.
+ *         Used in advertisement packets.
+ */
+#define GADGETS_UUID_SERVICE_SHORT 0x03, 0xFE
+
+/** @brief UUID of the TX Characteristic. */
+#define GADGETS_UUID_GADGETS_TX_CHAR \
+	BT_UUID_128_ENCODE(0xF04EB177, 0x3005, 0x43A7, 0xAC61, 0xA390DDF83076)
+
+/** @brief UUID of the RX Characteristic. */
+#define GADGETS_UUID_GADGETS_RX_CHAR \
+	BT_UUID_128_ENCODE(0x2BEEA05B, 0x1879, 0x4BB4, 0x8A2F, 0x72641F82420B)
+
+#define BT_UUID_GADGETS_SERVICE BT_UUID_DECLARE_128(GADGETS_UUID_SERVICE)
+#define BT_UUID_GADGETS_RX BT_UUID_DECLARE_128(GADGETS_UUID_GADGETS_RX_CHAR)
+#define BT_UUID_GADGETS_TX BT_UUID_DECLARE_128(GADGETS_UUID_GADGETS_TX_CHAR)
+
+
+/** @brief Gadgets Service result codes. */
+enum gadget_result {
+	/** Gadget result code: success. */
+	GADGETS_RESULT_CODE_SUCCESS = 0x00,
+	/** Gadget result code: unknown error. */
+	GADGETS_RESULT_CODE_UNKNOWN = 0x01,
+	/** Gadget result code: unsupported error. */
+	GADGETS_RESULT_CODE_UNSUPPORTED = 0x03,
+};
+
+/** @brief Gadgets Service stream types.
+ *         See Gadgets documentation for details:
+ *         https://developer.amazon.com/en-US/docs/alexa/alexa-gadgets-toolkit/packet-ble.html#streams
+ */
+enum stream_id {
+	/** Control stream ID */
+	STREAM_ID_CONTROL = 0b0000,
+	/** Alexa stream ID */
+	STREAM_ID_ALEXA = 0b0110,
+	/** OTA stream ID */
+	STREAM_ID_OTA = 0b0010,
+};
+
+/** @brief Callback type for data received.
+ *         Returns false if type is unsupported.
+ */
+typedef bool (*gadgets_stream_cb_t)(
```

Comment:
bt_gagdets_stream_cb_t

bt_gadgets_(stream?)_sent_cb_t

bt_gadgets_ccc_update_cb_t (gatt.h header use CCC so probable best to stick with that).

Reply:
Updated as suggested